src/Build/BackEnd/Client/MSBuildClientExitType.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+namespace Microsoft.Build.Execution
+{
+    public enum MSBuildClientExitType
+    {
+        /// <summary>
+        /// The MSBuild client successfully processed the build request.
+        /// </summary>
+        Success,
+        /// <summary>
+        /// Server is busy.
+        /// </summary>
+        ServerBusy,
+        /// <summary>
+        /// Client was unable to connect to the server.
+        /// </summary>
+        ConnectionError,
```

Comment:
is this tracking both unable to connect & connection dropped?  Does it matter to distinguish?

Reply:
Yes, there is a difference in behavior in these two cases. If not connected, than the regular build would be invoked as a fallback. It is some problem, it should not normally happen. But we are in the beginning and can afford just to call a regular build right away. If the connection has been dropped, then most probably it happen after some time of build execution (example: customer killed the server process). It is covered with `Unexpected` status. Then the fallback would not be called.

Reply:
I think about should we add any warning about this happening. I do not want to make it error. We fallback to normal build often - when the server is busy is one case. And we might as well do it in this case too.